writers/streamingbatchwriter/streamingbatchwriter_test.go

```diff
@@ -280,6 +284,65 @@ func TestStreamingBatchTimeout(t *testing.T) {
 	}
 }
 
+func TestStreamingBatchNoTimeout(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	ch := make(chan message.WriteMessage)
+
+	testClient := newClient()
+	wr, err := streamingbatchwriter.New(testClient, streamingbatchwriter.WithBatchTimeout(0), streamingbatchwriter.WithBatchSizeRows(2))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	errCh := make(chan error)
+	go func() {
+		errCh <- wr.Write(ctx, ch)
+	}()
+
+	table := schema.Table{Name: "table1", Columns: []schema.Column{{Name: "id", Type: arrow.PrimitiveTypes.Int64}}}
+	record := array.NewRecord(table.ToArrowSchema(), nil, 0)
+	ch <- &message.WriteInsert{
+		Record: record,
+	}
+	time.Sleep(50 * time.Millisecond)
+
+	if l := testClient.MessageLen(messageTypeInsert); l != 0 {
+		t.Fatalf("expected 0 insert messages, got %d", l)
+	}
+
+	time.Sleep(2 * time.Second)
+
+	if l := testClient.MessageLen(messageTypeInsert); l != 0 {
+		t.Fatalf("expected 0 insert messages, got %d", l)
+	}
+
+	ch <- &message.WriteInsert{
+		Record: record,
+	}
+	ch <- &message.WriteInsert{
+		Record: record,
+	}
+
+	time.Sleep(time.Millisecond * 250)
```

Comment:
Any way we can avoid the sleep here? Or at least sleep a smaller amount of time at regular intervals, then time out after some period of time. That should make it less flakey on CI systems where there can be resource contention

Reply:
We could use a fake time object of sorts to emulate timeout conditions in the future. batchingwriter also has some time.Sleeps.